### PR TITLE
play kube selinux label issue

### PR DESCRIPTION
**Pull request by st1971**
play kube function not respecting selinux options in kube yaml, all options were
being mapped to role.

fixes issue 8710

Signed-off-by: Steven Taylor <steven@taylormuff.co.uk>

### CODE REVIEW

**Comment by rhatdan** [comment]
Nice catch.
LGTM
Could you add a test.  Basically specify an SELinux label, and then create a kube container, and make sure the label is correct.

**Comment by rhatdan** [comment]
Nice catch.
LGTM
Could you add a test.  Basically specify an SELinux label, and then create a kube container, and make sure the label is correct.

@containers/podman-maintainers PTAL

**Comment by mheon** [comment]
Code changes LGTM

**Comment by mheon** [comment]
Would like to get this in for 3.0 final

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by st1971** [comment]
@rhatdan have added an additional test case to play_kube_test to test this, it's not pretty but validates that the process label is set correctly

**Comment by zhangguanzhang** [comment]
Fixes: https://github.com/containers/podman/issues/8710

File: test/e2e/play_kube_test.go [hunk 1]
```diff
@@ -803,6 +846,21 @@ var _ = Describe("Podman play kube", func() {
...
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube.ExitCode()).To(Equal(0))
+
+		inspect := podmanTest.Podman([]string{"inspect", "label-pod-test", "--format", "'{{ .ProcessLabel }}'"})
+		inspect.WaitWithDefaultTimeout()
+		label := inspect.OutputToString()
+
+		Expect(label).To(ContainSubstring("nconfined_u:system_r:spc_t:s0"))
```
Comment (rhatdan) [inline test/e2e/play_kube_test.go:861]: Missing a `u` unconfined_u

**Comment by rhatdan** [comment]
LGTM
Lets get this in and back ported to 3.0.

File: test/e2e/play_kube_test.go (hunk 1 above)
Comment (st1971) [reply test/e2e/play_kube_test.go]: sorry my bad have just correct

File: test/e2e/play_kube_test.go [hunk 2]
```diff
@@ -803,6 +846,21 @@ var _ = Describe("Podman play kube", func() {
 
 	})
 
+	It("podman play kube fail with custom selinux label", func() {
```
Comment (Luap99) [inline test/e2e/play_kube_test.go:849]: You have to add this to get pass the CI:
```golang
if !selinux.GetEnabled() {
	Skip("SELinux not enabled")
}
```
And make sure you import `github.com/opencontainers/selinux/go-selinux` at the top.

The test doesn't work on systems without selinux support like ubuntu.

Comment (rhatdan) [inline test/e2e/play_kube_test.go:849]: You need to add.
```
	if !selinux.GetEnabled() {
		Skip("SELinux not enabled")
       }
```

**Review by rhatdan — APPROVED** [review]

File: test/e2e/play_kube_test.go (hunk 2 above)
Comment (st1971) [reply test/e2e/play_kube_test.go:850]: sorry hadn't considered cases where selinux is not enabled, have updated the test

Comment (st1971) [reply test/e2e/play_kube_test.go:850]: sorry hadn't considered cases where selinux is not enabled, have updated the test

**Comment by rhatdan** [comment]
Thanks @st1971

**Pull request merged**